src/mlpack/tests/cf_test.cpp

```diff
@@ -37,6 +43,66 @@ using namespace mlpack;
 using namespace mlpack::cf;
 using namespace std;
 
+// Get train and test dataset. We save datasets as static variables
+// to avoid reloading and regenerating datasets.
+static void GetDatasets(arma::mat& dataset, arma::mat& savedCols)
+{
+  // Whether datasets have been initialized.
+  static bool datasetsInitialized = false;
+
+  static arma::mat initDataset;
+  static arma::mat initSavedCols(3, 50);
+
+  // If datasets have been initialized, just pass them to parameters.
+  if (datasetsInitialized)
```

Comment:
Are you sure this makes sense? I am not certain, but I think `ctest` runs each test as a separate process. Moreover, it isn't safe to use`static` variables with multi-threading, I guess it is better to avoid them at all even if we aren't going to use multi-threading at all.

Reply:
@lozhnikov I am not familiar with CTest so I searched for some relevant documents. CTest can run different tests in parallel, but all test cases in the same test suite still run in serial. And yes `static` variables are not safe in multithreading context. But as we are not using multithreading here, I think it would be fine to use static variables.
What do you think?

Reply:
Well, I measured the time of the test on my system. The whole `CFTest` takes 4.5 sec and 38.37 sec on my system in release and debug mode respectively. I tried to comment out the whole `if` statement and the difference is 0.3 sec and 1.22 sec in release and debug mode respectively. Thus, the difference is relatively small in comparison with the whole `CFTest`, so I suggest to remove static variables.

Reply:
I didn't expect the time difference to be so small. Then we should remove static variables. Thanks for testing the time difference!